gm4_block_compressors/data/gm4_block_compressors/functions/item/compress.mcfunction

```diff
@@ -0,0 +1,23 @@
+#@s = item entity to be compressed
+#run from block_compressors:item
+
+#copy existing data for restoration at decompression
+data modify storage gm4_block_compressors:temp/item_stack Item.tag.gm4_precompression_tag set from storage gm4_block_compressors:temp/item_stack Item.tag
+
+#if no tag data to save, make a note
+execute unless data storage gm4_block_compressors:temp/item_stack Item.tag run data modify storage gm4_block_compressors:temp/item_stack Item.tag.gm4_precompression_tag set value "no_tag_data_to_preserve"
```

Comment:
I know you adapted this from the current version of Block Compressors, but whilst you're updating this we should consider standardizing it.
This is a weird and non-standard way of introducing this custom tag. The NB tag should be under a tag named after the modules namespace. That would also prevent the tag from being of an ambiguous type. My suggest is to the following structure:
`gm4_block_compressors:{has_sealed_tag:<1/0b>,preserved_tag:{<data>}}`
The tag `has_sealed_data` might not even be necessary.

Now to the ugly: This would require implementing an update path for old items, but we already have one implemented in `decompress.mcfunction`, so that shouldn't be too bad.